synapseclient/pool_provider.py

```diff
@@ -0,0 +1,40 @@
+"""
+
+A thin wrapper around multiprocessing.dummy that allows users to use single thread. This module allow us to simplify the
+logic of any method that can run with either multiple threads or single thread.
+
+To use this wrapper with multiple threads::
+    pool = PoolProvider()
+    try:
+        pool.map(function, iterable)
+    finally:
+        pool.terminate()
+
+To use this wrapper for single thread::
+    pool = PoolProvider(single_threaded=True)
+
+"""
+from multiprocessing.dummy import Pool
+
+
+DEFAULT_POOL_SIZE = 8
+
+
+class SingleThreadPool:
+
+    def __init__(self):
+        pass
+
+    def map(self, func, iterable):
+        for item in iterable:
+            func(item)
+
+    def terminate(self):
+        pass
+
+
+def PoolProvider(single_threaded=False):
```

Comment:
snake case this and rename to get_pool()

this is currently doing the same thing that we do with table which hides the fact that it's not actually a classs